File: packages/core/rules/test/supply-chain/positive.py
# Test cases that SHOULD be detected by supply-chain rules

import subprocess

# mcp-python-subprocess-shell - should detect
subprocess.call(user_command, shell=True)
subprocess.run(user_input, shell=True)
subprocess.Popen(cmd, shell=True)

# mcp-python-exec-eval - should detect
exec(user_code)
eval(user_expression)

File: packages/core/rules/test/supply-chain/positive.js
// Test cases that SHOULD be detected by supply-chain rules

const { exec, execSync } = require('child_process');
const vm = require('vm');

// mcp-suspicious-exec-package - should detect
exec(userCommand);
execSync(userInput);

// mcp-suspicious-eval - should detect
eval(userCode);

// mcp-suspicious-vm-runInNewContext - should detect
vm.runInNewContext(untrustedCode);

// mcp-dynamic-import - should detect
require(userModule);
import(userPackage);
